### PR TITLE
ForClause var shadowing resolve fix

### LINKED ISSUE #1752: No new variables inspection should account for variable shadowing rules
The code below can run but the go plugin says it contains an error:

```go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	ShouldAllowParameterShadowing("param")
}

func ShouldAllowParameterShadowing(i string) {
	fmt.Println(i)

	for i := 0; i < 3; i++ { // Inspection error "No new variables on left side of :="
		fmt.Println(strconv.Itoa(i))
	}
}
```

But it will run correctly and output:
```
$ go run main.go
param
0
1
2
```

### PR DESCRIPTION
**Pull request by eugene-auduchinok**
A fix for wrong variable resolve in for clause. I hope it didn't break anything not covered with tests.
Fixes #1752.

### CODE REVIEW

**Comment by eugene-auduchinok** [comment]
Squashed in a single commit.

File: testData/highlighting/shortVars.go [hunk 1]
```diff
@@ -65,4 +65,10 @@ func Test2() (err error) {
     }
     
     return err
-}
\ No newline at end of file
+}
+
+func <warning>shadowing</warning>(i int) {
```
Comment (ignatov) [inline testData/highlighting/shortVars.go]: It's better to use _ as a name for such functions.

File: src/com/goide/psi/impl/GoVarProcessor.java [hunk 2]
```diff
@@ -35,6 +36,7 @@ public GoVarProcessor(@NotNull PsiElement origin, boolean completion) {
   public GoVarProcessor(@NotNull PsiElement requestedName, @NotNull PsiElement origin, boolean completion, boolean delegate) {
     super(requestedName, origin, completion);
     myImShortVarDeclaration = PsiTreeUtil.getParentOfType(origin, GoShortVarDeclaration.class) != null && !delegate;
+    myImInForClause = PsiTreeUtil.getParentOfType(origin, GoForClause.class) != null;
```
Comment (ignatov) [inline src/com/goide/psi/impl/GoVarProcessor.java]: What's about if/switch/... statements?

File: testData/highlighting/shortVars.go (hunk 1 above)
Comment (eugene-auduchinok) [inline testData/highlighting/shortVars.go]: Fixed and amended to the last commit.

File: src/com/goide/psi/impl/GoVarProcessor.java (hunk 2 above)
Comment (eugene-auduchinok) [inline src/com/goide/psi/impl/GoVarProcessor.java]: I've checked this and it works without these checks.

Comment (ignatov) [inline src/com/goide/psi/impl/GoVarProcessor.java]: Please add a test for if statement as well.

Comment (ignatov) [inline src/com/goide/psi/impl/GoVarProcessor.java]: And for other cases too.

**Comment by ignatov** [comment]
Looks good. So, all those changes should be squashed into one single commit.

File: src/com/goide/psi/impl/GoVarProcessor.java [hunk 3]
```diff
@@ -25,9 +25,9 @@
 
 public class GoVarProcessor extends GoScopeProcessorBase {
   private final boolean myImShortVarDeclaration;
-  private final PsiElement myParentGuard; 
+  private final PsiElement myParentGuard;
```
Comment (ignatov) [inline src/com/goide/psi/impl/GoVarProcessor.java]: Please revert this file. It doesn't have any changes.

**Comment by eugene-auduchinok** [comment]
Squashed.

**Pull request merged**